src/Illuminate/Routing/RouteDependencyResolverTrait.php

```diff
@@ -67,40 +76,6 @@ public function resolveMethodDependencies(array $parameters, ReflectionFunctionA
     }
 
     /**
-     * Attempt to transform the given parameter into a class instance.
-     *
-     * @param  \ReflectionParameter  $parameter
-     * @param  array  $parameters
-     * @param  array  $originalParameters
-     * @return mixed
-     */
-    protected function transformDependency(ReflectionParameter $parameter, $parameters, $originalParameters)
-    {
-        $class = $parameter->getClass();
-
-        // If the parameter has a type-hinted class, we will check to see if it is already in
-        // the list of parameters. If it is we will just skip it as it is probably a model
-        // binding and we do not want to mess with those; otherwise, we resolve it here.
-        if ($class && ! $this->alreadyInParameters($class->name, $parameters)) {
-            return $this->container->make($class->name);
-        }
-    }
-
-    /**
-     * Determine if an object of the given class is in a list of parameters.
-     *
-     * @param  string  $class
-     * @param  array  $parameters
-     * @return bool
-     */
-    protected function alreadyInParameters($class, array $parameters)
-    {
-        return ! is_null(Arr::first($parameters, function ($key, $value) use ($class) {
-            return $value instanceof $class;
-        }));
-    }
```

Comment:
The problem here is that if there are multiple same `$class`, the `$value` instance of `$class` in `$parameters` may be short of. (example: requiring three User class, but only provide two instance, so we mush make the third one)